File: app/src/main/java/com/example/dunzoclone/ViewHolders/SearchViewHolder.java
package com.example.dunzoclone.ViewHolders;

import android.annotation.SuppressLint;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.dunzoclone.DataModels.Product_Model;
import com.example.dunzoclone.R;

public class SearchViewHolder extends RecyclerView.ViewHolder {
    ImageView imageView;
    TextView product_name;
    TextView desc;
    TextView quantity;
    TextView price;
    TextView searchView;

    public SearchViewHolder(@NonNull View itemView) {
        super(itemView);
        imageView = itemView.findViewById(R.id.ivProductImage);
        searchView = itemView.findViewById(R.id.svSearch);
        product_name = itemView.findViewById(R.id.tvProductName);
        desc = itemView.findViewById(R.id.tvProductDesc);
        quantity = itemView.findViewById(R.id.tvQuantitySelector);
        price = itemView.findViewById(R.id.tvProductPrice);
    }


    public void setData(Product_Model product_model) {
        product_name.setText(product_model.getProduct_name());
        desc.setText(product_model.getProduct_desc());
        quantity.setText(product_model.getQuantity()+"");
        price.setText("₹ "+product_model.getPrice());
    }
}

File: app/src/main/java/com/example/dunzoclone/ViewHolders/SearchViewHolder.kt
//package com.example.dunzoclone.Activities
//
//import android.view.View
//import android.widget.ImageView
//import android.widget.TextView
//import androidx.recyclerview.widget.RecyclerView
//import com.example.dunzoclone.DataModels.Product_Model
//import com.example.dunzoclone.R
//
//class SviewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
//    var imageView: ImageView
//    var product_name: TextView
//    var desc: TextView
//    var quantity: TextView
//    var price: TextView
//    var searchView: TextView
//    fun setData(product_model: Product_Model) {
//        product_name.text = product_model.product_name
//        desc.text = product_model.product_desc
//        quantity.text = product_model.quantity.toString() + ""
//        price.text = "₹ " + product_model.price
//    }
//
//    init {
//        imageView = itemView.findViewById(R.id.ivProductImage)
//        searchView = itemView.findViewById(R.id.svSearch)
//        product_name = itemView.findViewById(R.id.tvProductName)
//        desc = itemView.findViewById(R.id.tvProductDesc)
//        TextView = itemView.findViewById(R.id.tvQuantitySelector)
//        price = itemView.findViewById(R.id.tvProductPrice)
//    }
//}